1_fetch.R

```diff
@@ -216,41 +217,43 @@ p1_targets_list <- list(
       mutate(COMID_refgages = as.character(nhdpv2_COMID))
   ),
   
-  # STATSGO SOIL Characteristics
-  # get selected child items nhdv2 STATSGO Soil Characteristics,
-  # including 1) texture and 2) layer attributes.
+  # Read in csv file containing the NHDPlusv2 segment/catchment attributes that 
+  # we want to download from ScienceBase:
   tar_target(
-    p1_selected_statsgo_sbid_children,
-    sbtools::item_list_children(sb_id = nhd_statsgo_parent_sbid) %>% 
-      Filter(function(x){str_detect(x[['title']],'Text|Layer')},
-             .)
+    p1_sb_attributes_csv,
+    '1_fetch/in/nhdv2_attributes_from_sciencebase.csv',
+    format = 'file'
   ),
   
-  # Download selected CONUS STATSGO datasets from Science base
+  # Read in and format segment/catchment attribute datasets from ScienceBase 
+  # note: use tar_group to define row groups based on ScienceBase ID; 
+  # row groups facilitate branching over subsets of the sb_attributes 
+  # table in downstream targets
   tar_target(
-    p1_download_statsgo_text_layer_zip,
-    lapply(p1_selected_statsgo_sbid_children,
-           function(x){download_sb_file(sb_id = x$id,
-                                        out_dir = '1_fetch/out/statsgo',
-                                        file_name = NULL,
-                                        overwrite_file = TRUE)}
-           ) %>%
-      unlist(),
-    format = 'file'
+    p1_sb_attributes,
+    read_csv(p1_sb_attributes_csv, show_col_types = FALSE) %>%
+      # parse sb_id from https link 
+      mutate(sb_id = str_extract(SB_link,"[^/]*$")) %>%
+      group_by(sb_id) %>%
+      tar_group(),
+    iteration = "group"
   ),
-
-  # Combine statsgo TEXT and Layer Attributes for CAT and TOT and filter to drb
+  
+  # Map over desired attribute datasets to download NHDv2 attribute data
+  # Note that a txt file within the BASIN_CHAR group returns a warning that
+  # it's skipped over. That is OK because the data is also included in a zip
+  # file on ScienceBase. The zip file is the version of the data that's being
+  # downloaded and saved here.
   tar_target(
-    p1_statsgo_soil_df,
-    sb_read_filter_by_comids(data_path = '1_fetch/out/statsgo',
-                             comid = p1_drb_comids_all_tribs$COMID,
-                             sb_comid_col = 'COMID',
-                             selected_cols_contains = c("KFACT","KFACT_UP","NO10AVE",
-                                                        "NO4AVE","SILTAVE","CLAYAVE",
-                                                        "SANDAVE",'WTDEP'),
-                             cbind = TRUE)
+    p1_sb_attributes_downloaded_csvs,
```

Comment:
Yes, I added a note in `1_fetch.R` above this target that we expect this warning message for this file only: `FTYPE_PCT_CONUS.txt`. That one is OK because there's a redundant zip file in that SB item that contains the same data. 

```r
  # Map over desired attribute datasets to download NHDv2 attribute data
  # Note that a txt file within the BASIN_CHAR group returns a warning that
  # it's skipped over. That is OK because the data is also included in a zip
  # file on ScienceBase. The zip file is the version of the data that's being
  # downloaded and saved here.
```

Reply:
Oh, no worries! I only pointed it out to indicate that we were expecting that particular warning 🙂